bundles/org.openhab.binding.ambientweather/src/main/java/org/openhab/binding/ambientweather/internal/handler/AmbientWeatherBridgeHandler.java

```diff
@@ -0,0 +1,233 @@
+                logger.debug("Bridge: Querying list of devices from ambient weather service");
+                response = HttpUtil.executeUrl("GET", url, DEVICES_API_TIMEOUT);
+                logger.trace("Bridge: Response = {}", response);
+            } catch (IOException e) {
+                // executeUrl throws IOException when it gets a Not Authorized (401) response
+                logger.debug("Bridge: Got IOException: {}", e.getMessage());
+                updateThingStatus(e.getMessage(), "Invalid API or application key");
+                rescheduleValidateKeysJob();
+                return;
+            } catch (IllegalArgumentException e) {
+                logger.debug("Bridge: Got IllegalArgumentException: {}", e.getMessage());
+                updateThingStatus(e.getMessage(), "Unable to get devices");
+                rescheduleValidateKeysJob();
+                return;
+            }
+            try {
+                // Got a response so the keys are good
+                DeviceJson[] stations = gson.fromJson(response, DeviceJson[].class);
+                logger.debug("Bridge: Application and API keys are valid with {} stations", stations.length);
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE, "Connecting to service");
+                // Start up the real-time API listener
+                listener.start(applicationKey, apiKey, gson);
+            } catch (JsonSyntaxException e) {
+                logger.debug("Bridge: Got JsonSyntaxException: {}", e.getMessage());
+                updateThingStatus(e.getMessage(), "Error parsing json response");
+                rescheduleValidateKeysJob();
+                return;
+            }
+        }
+    };
+
+    public AmbientWeatherBridgeHandler(Bridge bridge) {
+        super(bridge);
+        listener = new AmbientWeatherEventListener(this);
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN);
+        // If there are keys in the config, schedule the job to validate them
+        if (haveApplicationKey() && haveApiKey()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Awaiting key validation");
+            scheduleValidateKeysJob();
+        }
+    }
+
+    /*
+     * Check if an application key has been provided in the thing config
+     */
+    private boolean haveApplicationKey() {
+        String configApplicationKey = getConfigAs(BridgeConfig.class).applicationKey;
+        if (StringUtils.isEmpty(configApplicationKey)) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Missing application key");
+            return false;
+        }
+        applicationKey = configApplicationKey;
+        return true;
+    }
+
+    /*
+     * Check if an API key has been provided in the thing config
+     */
+    private boolean haveApiKey() {
```

Comment:
Nitpicking: hasApiKey?

Reply:
Yes to the nitpicking comment. 😄  Changed both method names.